docs/tutorials/cargo.md

```diff
@@ -0,0 +1,114 @@
+# Setting up your Cargo.toml file for ICU4X
+
+ICU4X makes heavy use of small crates and Cargo features in order to be highly modular. This tutorial is intended to help you set up a Cargo.toml file to download what you need for ICU4X.
+
+## Basic Cargo.toml with testdata
+
+The most basic Cargo.toml to get you off the ground is the following:
+
+```toml
+[package]
+name = "demo"
+version = "0.1.0"
+edition = "2021"
+publish = false
+
+[dependencies]
+icu = "1.0"
+icu_testdata = "1.0"
+```
+
+In your main.rs, you can use most functionality of ICU4X.
+
+[« Fully Working Example »](./cargo_tests/testdata)
+
+## Cargo.toml with experimental testdata
+
+If you wish to use an experimental feature, such as `icu_segmenter`, set up your Cargo.toml like this:
+
+```toml
+[package]
+name = "demo"
+version = "0.1.0"
+edition = "2021"
+publish = false
+
+[dependencies]
+icu = { version = "1.0", features = ["experimental"] }
+icu_testdata = { version = "1.0", features = ["icu_segmenter"] }
+```
+
+In your main.rs, you can now use experimental features.
+
+[« Fully Working Example »](./cargo_tests/experimental)
+
+## Cargo.toml with Buffer Provider
+
+If you wish to generate your own data in blob format and pass it into ICU4X, enable the "serde" Cargo feature as follows:
+
+```toml
+[package]
+name = "demo"
+version = "0.1.0"
+edition = "2021"
+publish = false
+
+[dependencies]
+icu = { version = "1.0", features = ["serde"] }
+icu_provider_blob = "1.0"
+```
+
+To generate your blob data, first run `cargo build` before you run `icu4x-datagen`.
+
+[« Fully Working Example »](./cargo_tests/buffer)
+
+## Cargo.toml with Baked Provider
+
+If you wish to bake in data auto-generated in your build.rs, you need to add additional dependencies as follows:
```

Comment:
issue: I'm a bit wary of recommending people call datagen (which is super heavy) in a build script without adding all the caveats that it is super heavy and linking to a dedicated tutorial about the various options here.

Reply:
also it's not normal for build scripts to hit the network, and in particular people using non-cargo build systems _hate_ this (but it's been very abnormal in Rust even before that. People don't like nondeterminism in their build). I'm not sure we should be recommending this model, especially since most users will want to generate-once-and-check-in.

Reply:
CC @robertbastian 

Valid points. It _is_ super convenient though that databake-datagen is capable of being run in build.rs. There are examples of build.rs being _heavy_, so I'm not super worried about that; actually, this build.rs is fairly fast given a limited set of keys and locales. Being _non-deterministic_ is a bit trickier; at least worth a note that you probably don't want to do this unless you are a leaf crate.

Also, this model is already suggested in passing in data_management.md, and I think we've discussed it before. I'm not intending to introduce anything new here; just adding an example of something you can already do.

Reply:
Yeah I think we should _document_ this approach somewhere but I do not think we should be endorsing it too much. It's fine for now, perhaps we can clean this up in a followup.

Reply:
Well, the network only gets hit if you call `SourceData::with_{icuexport}_{tag,latest}`, which clearly document that they're downloading data. If you don't use those, datagen is offline.